source/native/symbulation_sgp.cc

```diff
@@ -26,48 +27,67 @@ int symbulation_main(int argc, char *argv[]) {
   emp::Random random(config.SEED());
 
   TaskSet task_set = LogicTasks;
-
-  if (config.TASK_TYPE() == 0){
-     task_set = SquareTasks;
-  }else if (config.TASK_TYPE() == 1){
+  if (config.TASK_TYPE() == 0) {
+    task_set = SquareTasks;
+  } else if (config.TASK_TYPE() == 1) {
     task_set = LogicTasks;
   }
-
   SGPWorld world(random, &config, task_set);
 
   worldSetup(&world, &config);
   world.CreateDataFiles();
+  // Print some debug info for testing purposes
+  std::string file_ending = "_SEED" + std::to_string(config.SEED()) + ".data";
+
+  //emp::SignalKey OnAnalyzePopulation(const std::function<void()> & fun) {
+  //  return on_analyze_population_sig.AddAction(fun);
+  //}
+  world.OnAnalyzePopulation([&](){
+    emp::vector<CPU> cpus = {};
+    emp::vector<std::pair<emp::Ptr<Organism>, size_t>> dominant_organisms =
+      world.GetDominantInfo();
+    for (auto pair : dominant_organisms) {
+      auto sample = pair.first.DynamicCast<SGPHost>();
+      cpus.push_back(sample->GetCPU());
+      if(sample->HasSym()){
+        cpus.push_back(sample->GetSymbionts().front().DynamicCast<SGPSymbiont>()->GetCPU());
```

Comment:
I don't think you should have the hosts and symbionts mixed together for diversity analysis, you should get the diversity metrics of each population seperately